src/gp_instance.cpp

```diff
@@ -86,3 +130,805 @@ void GPInstance::ProcessLoadedTrees() {
   sbn_parameters_.resize(index);
   sbn_parameters_.setOnes();
 }
+
+void GPInstance::CreateAndInsertNode(const Bitset &subsplit)
+{
+  if (!subsplit_to_index_.count(subsplit)) {
+    size_t id = dag_nodes_.size();
+    subsplit_to_index_[subsplit] = id;
+    dag_nodes_.push_back(std::make_shared<DAGNode>(id, subsplit));
+  }
+}
+
+void GPInstance::AddChildrenSubsplits(const Bitset &subsplit,
+                                      std::deque<Bitset> &q,
```

Comment:
Especially given that `q` appears with a totally different meaning elsewhere, could we use a more descriptive name for it?

Reply:
Done. Changed it from `q` - > `subsplit_queue`.